inference-engine/cmake/ie_parallel.cmake

```diff
@@ -7,6 +7,7 @@ function(set_ie_threading_interface_for TARGET_NAME)
         find_package(TBB COMPONENTS tbb tbbmalloc)
         set("TBB_FOUND" ${TBB_FOUND} PARENT_SCOPE)
         set("TBB_IMPORTED_TARGETS" ${TBB_IMPORTED_TARGETS} PARENT_SCOPE)
+        set("TBB_VERSION" ${TBB_VERSION} PARENT_SCOPE)
```

Comment:
may we need to put them in cache? In this case we don't need play with PARENT_SCOPE in other files.

Reply:
Global cache is not always good idea. Particularly for this case we have a problem multiple TBB::tbb target definition only when call file package from one cmake script. If try it form different places cmake keeps this TBB import targets locally and there are no conflicts.

So if we change TBB_FOUND to cache it will break other places where TBB is used.

Other concern. As you know, cache are stored separately, and will be reused for next cmake reload. For that case we will miss TBB target forever till next reset cache and reload.

Reply:
Other question WHY did you move find_package inside a function? Was it request from @mryzhov?

Previously it was exported in root cmake file and will be available for each subfolders in OV project. But not each subfolder try to create it own version of tbb import.

Reply:
it was requested by @alalek because for 3rd party component we don't need to find TBB since TBB it's a private IE dependency.

E.g. when OpenCV is build against IE, InferenceEngineConfig.cmake finds TBB, but does not use it (it can use it only if set_threading_interface is called for some targets)

Reply:
To avoid issues with function's scope it makes sense to convert it into a macro (or split into a macro part with `find_package()` and a function with internal logic).

Reply:
during skype discussion we also agreed with @AlexPeskov that we also need to find TBB only in IE paths (and we need to set TBB_DIR / TBBROOT for that before calling `ie_parallel.cmake`), because currently find TBB can find any TBB on the system, because nobody sets proper location when we use prebuild OpenVINO package.

Reply:
@AlexPeskov are you going to apply changes we agreed?